networking_cisco/plugins/cisco/db/l3/l3_router_appliance_db.py

```diff
@@ -223,6 +223,11 @@ def create_router(self, context, router):
             # Ensure create spec is compliant with any HA
             ha_spec = self._ensure_create_ha_compliant(r, router_type)
         auto_schedule, share_host = self._ensure_router_scheduling_compliant(r)
+        driver = self._get_router_type_driver(context,
+                                              router_type_id)
+        if driver:
+            router_ctxt = driver_context.RouterContext(router)
+            driver.create_router_precommit(context, router_ctxt)
```

Comment:
overall looks good to me. Some questions:
1. I wonder what's the convention to call those pre and post-commit(). In this case, only create_router_precommit() is called but not the postcommit(). In some other case its the other way around. Then in the FIP cases, both pre and post commits are called.....So I'm getting a bit confused here.